Userland/Libraries/LibGfx/ICC/BinaryWriter.cpp

```diff
@@ -569,7 +569,7 @@ static ErrorOr<ByteBuffer> encode_xyz(XYZTagData const& tag_data)
     return bytes;
 }
 
-static ErrorOr<ByteBuffer> encode_tag_data(TagData const& tag_data)
+static Optional<ErrorOr<ByteBuffer>> encode_tag_data(TagData const& tag_data)
```

Comment:
`ErrorOr<Optional<ByteBuffer>>` please

Reply:
Isn't that something else? An `ErrorOr<Optional<ByteBuffer>>` is an error or maybe a byte buffer, while this maybe make the attempt to create a byte buffer, which can fail.

At least in my head those are two different things!

(Or is this something from the coding style document I had missed?)

Reply:
I wouldn't assign different meaning to them, both are "fallible function that may or may not return a byte buffer".
Nested `ErrorOr` is just pretty unusual and it seems nicer to just `TRY` that away upfront, even if it technically doesn't even apply in the `Optional` case.

There's lots of precedent for `ErrorOr<Optional<T>>`: https://github.com/search?q=repo%3ASerenityOS%2Fserenity%20%22ErrorOr%3COptional%22&type=code

But AFAICT none for `Optional<ErrorOr<T>>`: https://github.com/search?q=repo%3ASerenityOS%2Fserenity+%22Optional%3CErrorOr%22&type=code

Reply:
I guess the TL;DR is "arbitrary unwritten style preference" 🙃